### PR TITLE
add E2E test cases for all demo page features

### PR DESCRIPTION
**Pull request by liunate**
- Add E2E test cases for all demo page features(at least to features explained on demo page)
- Small refactor to each test case containing element alias name for better readability

### CODE REVIEW

File: cypress/integration/focus-trap-demo.spec.js [hunk 1]
```diff
@@ -55,4 +54,73 @@ describe('<FocusTrap> component', () => {
...
+
+  it('Can click outside of trap to deactivate and click carries through  ', () => {
+    cy.get('#demo-special-element')
+      .as('testRoot');
+
+    // activate trap
+    cy.get('@testRoot').findByRole('button', { name: 'activate trap' }).click();
+
+    // click outside to deactivate trap and also click carries through
+    cy.findByRole('button', { name: 'pass thru click' }).click().should('be.focused');
```
Comment (liunate) [inline cypress/integration/focus-trap-demo.spec.js:112]: I was trying to just verify `document.activeElement` when click outside, but `Chrome` and `Firefox` has different behavior on clicking outside.

`Chrome` focuses on lastly focused element while `Firefox` focusing on `<body>`. Also it's not clear if it's just trap losing focus(deactivate) and also click passes thru. 

So I update the demo page a little at [new button](https://github.com/focus-trap/focus-trap-react/pull/103/files#diff-a4c3e7d6c4e1076bff9fb47c75a32f9cR43) so it clearly shows it's not only focus is freed but also click passes thru.

File: cypress/integration/focus-trap-demo.spec.js [hunk 2]
```diff
@@ -55,4 +54,73 @@ describe('<FocusTrap> component', () => {
...
+
+  it('Focus on element with "autofocus" attribute than the 1st tab order element within mounted trap children', () => {
+    cy.get('#demo-autofocus')
+      .as('testRoot');
+
+    // activate trap
+    cy.get('@testRoot').findByRole('button', { name: 'activate trap' }).click();
+
+    // element with "autofocus" attribute is focused
+    cy.findByTestId('autofocus-el').should('be.focused');
```
Comment (liunate) [inline cypress/integration/focus-trap-demo.spec.js:124]: demo page is updated a little to have some [easier selector](https://github.com/focus-trap/focus-trap-react/pull/103/files#diff-d7e657e37f7e3c3155b02d20d9560b6aR41) for e2e test.

File: cypress/integration/focus-trap-demo.spec.js [hunk 3]
```diff
@@ -1,19 +1,18 @@
-// TODO: maybe we just need one spec file for all
 describe('<FocusTrap> component', () => {
-  it('By default focus first element in its tab order and trap focus within its children', () => {
-    cy.visit('index.html');
+  beforeEach(() => cy.visit('index.html'));
 
+  it('By default focus first element in its tab order and trap focus within its children', () => {
     cy.get('#demo-defaults')
-      .as('trap');
+      .as('testRoot');
```
[the remark this answers is not in the document]
Comment (stefcameron) [reply cypress/integration/focus-trap-demo.spec.js:6]: Definitely! 👍

**Review by stefcameron — APPROVED** [review]
Nice work on this! Thanks for completing the set of tests. This adds a really nice level of coverage on the project which `focus-trap` also desperately needs. It has 0 tests at the moment, but it does have a similar demo page, so I'm going to setup Cypress there. Then I'll see if I can get help to gradually add all coverage for all the test cases in the demo, just like here. Your work here will make it easy for me to setup Cypress there! 🎉

**Comment by stefcameron** [comment]
No changeset needed for this: PR doesn't add new features (just ensures quality of new ones to come!!).

**Pull request merged**